### DISCUSSION TITLE
How can I calculate the total bandwidth usage of a viewer?

### DISCUSSION THREAD

**Discussion by tremzo** [ant-media/Ant-Media-Server#5860] [FAQs] [answered]
I need to be able to calculate how much bandwidth a viewer is using.

**Comment by akiziltoprak** [comment]
Hey

Please use https://antmedia.io/cost-calculator/ by entering the values including concurrent publisher and viewer count, plus also video resolution that you will use for ingesting and when the calculation is made, data bandwidth use will be provided as total in the results.

**Comment by lastpeony** [comment, marked as answer]
Hi,
You can obtain the average bitrate per second from the subscriber object. Multiply this bitrate by the subscriber's playing time to calculate the total usage. To implement this, start by creating a subscriber for the stream using the following API endpoint: [addSubscriber](https://antmedia.io/rest/#/BroadcastRestService/addSubscriber).

While playing, pass the subscriberId to the play request using the SDK. You can achieve this by passing the subscriberId to the `webrtcAdaptor.play()` method or as a URL parameter like this: `http://localhost:5080/WebRTCAppEE/play.html?id=teststream&subscriberId=testsubscriber`.

When a user begins playing through the provided URL, the subscriber stats object will store various statistics, including the average bitrate for video and audio.

![image](https://github.com/ant-media/Ant-Media-Server/assets/24750845/d53e8ba6-db90-4f5d-adf3-58765398f8e8)

The avgVideoBitrate represents the total number of bits (not bytes) sent from the server to the user every second.

To retrieve subscriber stats for a specific streamId, use the following GET request: [listSubscriberStatsV2](https://antmedia.io/rest/#/BroadcastRestService/listSubscriberStatsV2).

This method allows you to gather all subscriber statistics for a particular stream. Afterwards, you can obtain the average video bitrate and average audio bitrate for your spesific subscriber. The connectionEvents object provides information about when subscribers started and stopped playing. By utilizing this data, you can calculate their total playing time in seconds. Multiplying this duration by the average video bitrate or average audio bitrate enables you to determine the total bandwidth consumption for a specific viewer.
